src/workerd/jsg/jsg.h

```diff
@@ -2246,42 +2246,72 @@ JS_TYPE_CLASSES(V)
 #undef V
 
 class DOMException;
+class ExternalMemoryAdjustment;
+class ExternalMemoryPtr: public kj::Refcounted {
+  struct Impl;
+
+ public:
+  ExternalMemoryPtr(): maybeInner(kj::none) {}
+
+  ExternalMemoryPtr(v8::Isolate* isolate, v8::ExternalMemoryAccounter* accounter) {
+    maybeInner.emplace(isolate, accounter);
+  }
+
+  bool isIsolateAlive();
+  int64_t getPendingMemoryUpdate();
+  ExternalMemoryAdjustment getAdjustment(size_t amount);
+  ~ExternalMemoryPtr();
+
+ private:
+  kj::Own<ExternalMemoryPtr> addRef();
+  void reset();
+  void maybeDeferAdjustment(ssize_t amount);
+  struct Impl {
+    v8::Isolate* isolate;
+    v8::ExternalMemoryAccounter* externalMemoryAccounter;
+  };
+
+  kj::Maybe<Impl> maybeInner;
+
+  friend class ExternalMemoryAdjustment;
+  friend class IsolateBase;
+};
+
+// Used to save a reference to an isolate that is responsible for external memory usage.
+// getAdjustment() can be invoked at any time to create a new RAII adjustment object
+// pointing to this isolate
+using ExternalMemoryTarget = kj::Own<ExternalMemoryPtr>;
```

Comment:
This alias may be a bit confusing/counter-productive. Whenever we have a `kj::Own<T>` it's best for that to be explicit so that folks know how to handle it. Just seeing `ExternalMemoryTarget` they may not realize that it's actually an own.